lib/ProductOpener/Display.pm

```diff
@@ -6797,8 +6796,10 @@ $Lang{warning_gs1_company_prefix}{$lc}
 HTML
 ;
 	}
-
-	if (not has_tag($product_ref, "states", "en:complete")) {
+	if (defined $rev) {
+		$html .= display_rev_info($code, $rev);
+	}
+	if (not has_tag($product_ref, "states", "en:complete") and not defined $rev) {
```

Comment:
```suggestion
	elsif (not has_tag($product_ref, "states", "en:complete")) {
```
I'm always in doubt about precedence when I see things "like and not defined" without parenthesis.